Add Charmhub documentation (#45)
canonical/tls-certificates-operator: docs/index.md

> The TLS Certificates Operator charm is responsible for distributing certificates through relationships. Certificates are provided by the operator through Juju configs.

Reviewer: ```suggestion
The TLS Certificates Operator charm is responsible for distributing x.509 certificates in contexts where certificates are obtained through a manual process.
```

Author: Done.

## Usage

> To deploy the TLS Certificates Operator, all you need to do is run the following command, which will fetch the charm from [Charmhub](https://charmhub.io/tls-certificates-operator) and deploy it to your model:

Reviewer: ```suggestion
Deploy the charm:
```

Author: Done.

> ```shell
> juju deploy tls-certificates-operator
> ```

Reviewer: ```suggestion
```shell
juju deploy tls-certificates-operator
juju status --watch 1s
```
```

Author: Done.

> Juju will now fetch TLS Certificates Operator and begin deploying it to the local MicroK8s. This process can take several minutes depending on how provisioned (RAM, CPU, etc) your machine is. You can track the progress by running:
> ```shell
> juju status --watch 1s
> ```
> The deployment of the charm is completed when its status is in `Active/Idle` state.

Reviewer: ```suggestion
```

Author: Done.

> At this point you can relate the TLS Certificates Operator to your charm through the `tls-certificates` interface:

Reviewer: Relate to a requirer of TLS certificates:

Author: Done.

```shell
juju relate tls-certificates-operator <your-charm>
```

> Wait for `juju status --watch 1s` to show all applications/units as `active`.

Reviewer: ```suggestion
```

Author: Done.

| [Reference](https://charmhub.io/tls-certificates-operator/actions) </br> **Technical information** - specifications, APIs, architecture                           |

## Project and community

TLS Certificates Operator is a member of the Ubuntu family. It’s an open source project that warmly welcomes community projects, contributions, suggestions, fixes and constructive feedback.

* [Code of conduct](https://ubuntu.com/community/ethos/code-of-conduct)
* Contribute and report bugs to the [charm](https://github.com/canonical/tls-certificates-operator)

# Contents
| Level | Path                 | Navlink                                                                                                                                                |
|-------|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
| 1     | how-to               | [How-to guides]()                                                                                                                                      |

> | 2     | h-configure-per-unit | [Configure certificates per unit](/t/tls-certificates-operator-how-to-configure-per-unit-certificates/11612)                                           |

Reviewer: I don't especially like the `per unit` wording that we have all over the place. Users use this charm whenever they obtain their certificates in a manual process. The per unit aspect is mostly an implementation detail, not what people care about.

Author: I removed the `per unit` stuff and replaced with "Provisiong of certificates"
